code/mob/living/carbon/human/procs/Life.dm

```diff
@@ -1775,24 +1737,22 @@
 						boutput(src, "<span class='alert'><b>You feel [pick("terrible", "awful", "like shit", "sick", "numb", "cold", "really sweaty", "tingly", "horrible")]!</b></span>")
 						src.changeStatus("weakened", 3 SECONDS)
 
-		parent.setLastTask("status_updates blindness checks", src)
-
 		//todo : clothing blindles flags for less istypeing
 		if (getStatusDuration("blinded"))
 			src.blinded = 1
 
-		if (istype(src.glasses, /obj/item/clothing/glasses/))
+		else if (istype(src.glasses, /obj/item/clothing/glasses/))
```

Comment:
These things just set blinded to 1 conditionally, no need to check them if we already set blinded.